sdk/data_kusto/src/connection_string.rs

```diff
@@ -0,0 +1,336 @@
+    /// Denotes if MSI authorization should be used
+    pub msi_auth: Option<bool>,
+    pub msi_params: Option<&'a str>,
+    pub az_cli: Option<bool>,
+}
+
+impl<'a> PartialEq for ConnectionString<'a> {
+    fn eq(&self, other: &Self) -> bool {
+        self.data_source == other.data_source
+            && self.federated_security == other.federated_security
+            && self.user_id == other.user_id
+            && self.user_token == other.user_token
+            && self.password == other.password
+            && self.application_client_id == other.application_client_id
+            && self.application_key == other.application_key
+            && self.application_token == other.application_token
+            && self.application_certificate == other.application_certificate
+            && self.application_certificate_thumbprint == other.application_certificate_thumbprint
+            && self.authority_id == other.authority_id
+            && self.msi_auth == other.msi_auth
+            && self.msi_params == other.msi_params
+            && self.az_cli == other.az_cli
+    }
+}
+
+impl<'a> ConnectionString<'a> {
+    pub fn new(connection_string: &'a str) -> Result<Self, ConnectionStringError> {
+        let mut data_source = None;
+        let mut federated_security = None;
+        let mut user_id = None;
+        let mut user_token = None;
+        let mut password = None;
+        let mut application_client_id = None;
+        let mut application_token = None;
+        let mut application_key = None;
+        let mut application_certificate = None;
+        let mut application_certificate_thumbprint = None;
+        let mut authority_id = None;
+        let mut msi_auth = None;
+        let mut msi_params = None;
+        let mut az_cli = None;
+
+        let kv_str_pairs = connection_string
+            .split(';')
+            .filter(|s| !s.chars().all(char::is_whitespace));
+
+        for kv_pair_str in kv_str_pairs {
+            let mut kv = kv_pair_str.trim().split('=');
+            let k = match kv.next() {
+                Some(k) if k.chars().all(char::is_whitespace) => {
+                    return Err(ConnectionStringError::ParsingError {
+                        msg: "No key found".to_owned(),
+                    })
+                }
+                None => {
+                    return Err(ConnectionStringError::ParsingError {
+                        msg: "No key found".to_owned(),
+                    })
+                }
+                Some(k) => k,
+            };
+            let v = match kv.next() {
+                Some(v) if v.chars().all(char::is_whitespace) => {
+                    return Err(ConnectionStringError::MissingValue { key: k.to_owned() })
+                }
+                None => return Err(ConnectionStringError::MissingValue { key: k.to_owned() }),
+                Some(v) => v,
+            };
+
+            match k {
+                DATA_SOURCE_NAME => data_source = Some(v),
```

Comment:
Parsing connection strings are a bit more complex than that - the names aren't case sensitive, and some of them have a aliases.
Take a look at https://github.com/Azure/azure-kusto-node/blob/master/azure-kusto-data/source/connectionBuilder.ts

Reply:
added a basic function to support aliases for keys based on the link. Need to look into the implementation there a bit more, but i think we need to normalize a bit more via lower casing input strings.